tests/core/test_run.py

```diff
@@ -93,3 +93,47 @@ def test_run_versions(self, test_mp, request):
         # non-default version cannot be again set as un-default
         with pytest.raises(IxmpError):
             run2.unset_as_default()
+
+    def test_filter_run(self, test_mp, request):
+        test_mp = request.getfixturevalue(test_mp)
+        run1, _ = create_filter_test_data(test_mp)
+
+        res = test_mp.runs.tabulate(
+            default_only=False,
+            iamc={
+                "region": {"name": "Region 1"},
+            },
+        )
+        assert sorted(res["model"].tolist()) == ["Model 1"]
+        assert sorted(res["scenario"].tolist()) == ["Scenario 1"]
+
+        res = test_mp.runs.tabulate(
+            default_only=False,
+            iamc={
+                "region": {"name": "Region 3"},
+            },
+        )
+        assert sorted(res["model"].tolist()) == ["Model 1", "Model 1", "Model 2"]
+        assert sorted(res["scenario"].tolist()) == [
+            "Scenario 1",
+            "Scenario 1",
+            "Scenario 2",
+        ]
+
+        run1.set_as_default()
+        res = test_mp.runs.tabulate(
+            iamc={
+                "variable": {"name": "Variable 1"},
+                "unit": {"name__in": ["Unit 3", "Unit 4"]},
+            }
+        )
+        assert sorted(res["model"].tolist()) == ["Model 2"]
+        assert sorted(res["scenario"].tolist()) == ["Scenario 2"]
+
+        res = test_mp.runs.tabulate(
+            default_only=False,
+            scenario={"name": "Scenario 2"},
+        )
+
+        assert sorted(res["model"].tolist()) == ["Model 2"]
+        assert sorted(res["scenario"].tolist()) == ["Scenario 2"]
```

Comment:
Suggest to extend the test in the following way:
- Remove all datapoints for the run Model 1, Scenario 1 related to Region 3
- Repeat the assertion form above
  ```python
           res = test_mp.runs.tabulate(
             default_only=False,
             iamc={
                 "region": {"name": "Region 3"},
             },
         )
   ```
- This should now only return Model 2, Scenario 2

Reply:
@danielhuppmann thanks for raising this. I added a test as you proposed and only runs with existing datapoints are returned :+1: